searchlib/src/tests/query/word_alternatives/word_alternatives_test.cpp

```diff
@@ -0,0 +1,191 @@
+    void visit(RegExpTerm &) override { abort(); }
+    void visit(NearestNeighborTerm &) override { abort(); }
+    void visit(TrueQueryNode &) override { abort(); }
+    void visit(FalseQueryNode &) override { abort(); }
+    void visit(FuzzyTerm &) override { abort(); }
+    void visit(InTerm&) override { abort(); }
+};
+
+
+// Builds a tree with simplequery and checks that the results have the
+// correct concrete types.
+TEST(WordAlternativesTest, require_that_Simple_Query_Trees_Can_Be_Built) {
+    Node::UP node = createQueryTree<SimpleQueryNodeTypes>();
+    EXPECT_TRUE(bool(node));
+    DumpVisitor dumper;
+    node->accept(dumper);
+}
+
+struct MyWordAlternatives : WordAlternatives {
+    MyWordAlternatives(std::unique_ptr<TermVector> terms, const string& v, int32_t i, Weight w)
+      : WordAlternatives(std::move(terms), v, i, w)
+    {}
+    ~MyWordAlternatives() override;
+};
+
+
+MyWordAlternatives::~MyWordAlternatives() = default;
+
+struct MyQueryNodeTypes : SimpleQueryNodeTypes {
+    using WordAlternatives = MyWordAlternatives;
+};
+
+TEST(WordAlternativesTest, require_that_tree_can_be_replicated) {
+    Node::UP node = createQueryTree<SimpleQueryNodeTypes>();
+    EXPECT_TRUE(bool(node));
+    Node::UP new_node = QueryTreeCreator<MyQueryNodeTypes>::replicate(*node);
+    EXPECT_TRUE(bool(new_node));
+    DumpVisitor dumper;
+    new_node->accept(dumper);
+}
+
+TEST(WordAlternativesTest, require_that_tree_can_be_replicated_via_stack) {
+    Node::UP node = createQueryTree<SimpleQueryNodeTypes>();
+    string stackDump = StackDumpCreator::create(*node);
+    SimpleQueryStackDumpIterator iterator(stackDump);
+    Node::UP new_node = QueryTreeCreator<MyQueryNodeTypes>::create(iterator);
+    EXPECT_TRUE(bool(new_node));
+    DumpVisitor dumper;
+    new_node->accept(dumper);
+}
+
+TEST(WordAlternativesTest, require_that_blueprints_can_be_built) {
+    Node::UP node = createQueryTree<SimpleQueryNodeTypes>();
+    auto *a = dynamic_cast<search::query::And *>(node.get());
+    EXPECT_TRUE(bool(a));
+    auto *p = dynamic_cast<search::query::Phrase *>(a->getChildren()[1]);
+    EXPECT_TRUE(bool(p));
+    FakeSearchable fake_index;
+    auto w1r = FakeResult().doc(7).doc(8).doc(9).doc(10).doc(17).elem(0).len(7).pos(3);
+    auto w2r = FakeResult().doc(4).doc(5).doc(6).doc(23).elem(0).len(19).pos(11);
+    auto w3r = FakeResult().doc(2).doc(3).doc(23).elem(0).len(19).pos(12);
+    auto w4r = FakeResult().doc(17).elem(0).len(7).pos(4);
+    fake_index.addResult(view[2], str[3], w1r);
+    fake_index.addResult(view[2], str[4], w2r);
+    fake_index.addResult(view[2], str[5], w3r);
+    fake_index.addResult(view[2], str[6], w4r);
+    FakeRequestContext req_ctx;
+    FieldSpecList fields;
+    fields.add(FieldSpec(view[2], 42, 17));
+    auto bp = fake_index.createBlueprint(req_ctx, fields, *p);
+    EXPECT_TRUE(bool(bp));
+    printf("Got blueprint: '%s'\n", bp->asString().c_str());
+}
```

Comment:
consider running the search. Also consider specific inspector for the blueprint tree